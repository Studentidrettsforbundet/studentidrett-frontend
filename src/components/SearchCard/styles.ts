import { style, classes } from 'typestyle';
import colors from '../../styles/colors';



export const searchCard = classes(
    style({
        cursor: 'pointer',
        margin: '0 auto 16px',
        padding: '10px',
        maxWidth: '500px',
        backgroundColor: colors.white,
        filter: 'drop-shadow(0px 4px 4px rgba(0, 0, 0, 0.25))',
        borderRadius: '5px',
        $nest: {
            '& span': {
                color: colors.secondary,
                textAlign: 'center',
                marginTop: 'auto',
                marginBottom: 'auto'
            },
            '& h5': {
                color: colors.orange,
            }
        },
    }),
);