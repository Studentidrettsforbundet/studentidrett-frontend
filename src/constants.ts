export const BASE_URL = 'https://kundestyrt-nsi-backend.azurewebsites.net';

export const CLUB = 'clubs';
export const SPORT = 'sports';
export const TEAM = 'teams';
export const REGION = 'regions';
export const CITY = 'cities';
export const GROUP = 'groups';
export const SEARCH = 'search';

export const MIDTNORGE = 'Midtnorge';
export const SORLANDET = 'Sørlandet';
export const VESTLANDET = 'Vestlandet';
export const OSTLANDET = 'Østlandet';
export const NORDNORGE = 'Nordnorge';

export type cardType =
    | typeof CLUB
    | typeof SPORT
    | typeof TEAM
    | typeof REGION
    | typeof CITY
    | typeof GROUP
    | typeof SEARCH;
