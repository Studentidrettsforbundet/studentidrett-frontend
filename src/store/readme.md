# Redux

This folder contains all redux files.

Add this extension to your browser: https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd

## Description:

store.ts is the file that creates the redux store, as well as the combined reducer. All reducers needs to be added to the combined reducer, and to the combinedState

This folder contains a folder for each page of our application that needs redux.
The folders contains reducers, actions, action crators, types, interfaces, etc. related to pages.

## Example of folder structure

We have a page called "search". That page needs a reducer, some actions, action creators, and some interfaces. Below is an example of the folder structure

----/store
--------store.ts
--------/pages
-----------/search
-----------------searchReducer.ts
-----------------searchActions.ts
-----------------searchReducer.ts
