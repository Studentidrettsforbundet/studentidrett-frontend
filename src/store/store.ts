import { applyMiddleware, combineReducers, compose, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import { cityInitialState, cityReducer, cityState } from './pages/city/cityReducer';
import { clubInitialState, clubReducer, clubDetailReducer, clubState } from './pages/club/clubReducer';
import { groupInitialState, groupReducer, groupDetailReducer, groupState } from './pages/group/groupReducer';
import { regionInitialState, regionReducer, regionState } from './pages/region/regionReducer';
import { sportInitialState, sportReducer, sportDetailReducer, sportState } from './pages/sport/sportReducer';
import { teamInitialState, teamReducer, teamDetailReducer, teamState } from './pages/team/teamReducer';
import { searchBarInitialState, searchBarReducer, searchBarState } from './searchBar/searchBarReducer';
import { thunkInitialState, thunkReducer, thunkState } from './thunks/thunkReducer';
import { searchInitialState, searchReducer, searchState } from './pages/search/searchReducer';
import {
    questionInitialState,
    questionnaireReducer,
    questionnaireState,
} from './pages/questionnaire/questionnaireReducer';

declare global {
    interface Window {
        __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose;
    }
}

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

//TODO: add reducers interfaces
export interface combinedStateInterface {
    region: regionState;
    city: cityState;
    sport: sportState;
    sport_detail: sportState;
    club: clubState;
    club_detail: clubState;
    team: teamState;
    team_detail: teamState;
    group: groupState;
    group_detail: groupState;
    search_results: searchState;
    thunk: thunkState;
    searchBar: searchBarState;
    questionnaire: questionnaireState;
}

// TODO: add all pages state here
export const combinedState = {
    region: regionInitialState,
    city: cityInitialState,
    sport: sportInitialState,
    sport_detail: sportInitialState,
    club: clubInitialState,
    club_detail: clubInitialState,
    team: teamInitialState,
    team_detail: teamInitialState,
    group: groupInitialState,
    group_detail: groupInitialState,
    search_results: searchInitialState,
    thunk: thunkInitialState,
    searchBar: searchBarInitialState,
    questionnaire: questionInitialState,
};

// create store
const store = createStore(
    combineReducers({
        region: regionReducer,
        city: cityReducer,
        sport: sportReducer,
        sport_detail: sportDetailReducer,
        club: clubReducer,
        club_detail: clubDetailReducer,
        team: teamReducer,
        team_detail: teamDetailReducer,
        group: groupReducer,
        group_detail: groupDetailReducer,
        search_results: searchReducer,
        thunk: thunkReducer,
        searchBar: searchBarReducer,
        questionnaire: questionnaireReducer,
    }),
    combinedState,
    composeWithDevTools(applyMiddleware(thunk)),
);

export default store;
