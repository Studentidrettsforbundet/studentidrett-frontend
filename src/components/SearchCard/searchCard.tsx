import React from 'react';
import { CITY, CLUB, GROUP, SPORT, TEAM } from '../../constants';
import { searchInterface } from '../../interfaces';
import { Link } from 'react-router-dom';
import { resetFetchStatusesActionCreator } from '../../store/thunks/thunkActions';
import { useDispatch } from 'react-redux';
import { searchCard } from './styles';

interface searchCardInterface {
    label: string;
}

const translations: { [index: string]: string } = {
    [CITY]: 'By',
    [SPORT]: 'Idrett',
    [GROUP]: 'Gruppe',
    [TEAM]: 'Idrettslag',
    [CLUB]: 'Idrettsklubb',
};

const SearchCard = ({ label, name, id }: searchCardInterface & searchInterface): JSX.Element => {
    const dispatch = useDispatch();
    return (
        <React.Fragment>
            <Link
                to={`/${label}/${id}`}
                className={'unstyled_link'}
                onClick={() => dispatch(resetFetchStatusesActionCreator())}
            >
                <div className={searchCard}>
                    <h5 className="boldText">{name}</h5>
                    <span>{translations[label]}</span>
                </div>
            </Link>
        </React.Fragment>
    );
};

export default SearchCard;
