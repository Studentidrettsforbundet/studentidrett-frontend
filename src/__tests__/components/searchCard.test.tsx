import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import SearchCard from '../../components/SearchCard/searchCard';
import store from '../../store/store';
import { CITY } from '../../constants';

describe('regionCard', () => {
    test('renders correctly', () => {
        const tree = renderer.create(
            <Provider store={store}>
                <MemoryRouter>
                    <SearchCard id={'1'} name={'Test'} label={CITY} />
                </MemoryRouter>
            </Provider>,
        );
        expect(tree).toMatchSnapshot();
    });
});
