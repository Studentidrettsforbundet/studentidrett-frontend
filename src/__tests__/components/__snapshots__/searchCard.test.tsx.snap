// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`regionCard renders correctly 1`] = `
<div>
  <a
    className="unstyled_link"
    href="/cities/1"
    onClick={[Function]}
  >
    <div
      className="fq2301v"
    >
      <h5>
        Test
      </h5>
      By
    </div>
  </a>
</div>
`;
